internal/store/memory/store.go

```diff
@@ -175,12 +157,6 @@ func (s *Store) SyncAtInterval(ctx context.Context) {
 			case <-ticker.C:
 				s.mutex.Lock()
 
-				// Force each gateway to sync its state even though listeners
-				// on the gateway may not be marked as needing a sync right now
-				for _, gateway := range s.gateways {
-					gateway.needsSync = true
-				}
-
```

Comment:
The sync is not gated the way it was before, so there's no need to "force the sync gate open" as this was doing before. The loop just needs to call `sync`. As Andrew mentions in a comment below in this file, we'll be adding some dirty checking logic in a followup PR to prevent an endless loop of reconciliation.